### PR TITLE
Add release flow

### PR DESCRIPTION
**Pull request by maxrjones**
Builds off https://github.com/zarr-developers/VirtualiZarr/pull/133.

This PR adds a GitHub workflow to test distributions using TestPyPI and upload releases to PyPI.

- [ ] Closes #xxxx
- [ ] Tests added
- [ ] Changes are documented in `docs/releases.rst`
- [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          fi
+      - uses: actions/upload-artifact@v4
+        with:
+          name: releases
+          path: dist
+
+  test-built-dist:
+    needs: build-artifacts
+    runs-on: ubuntu-latest
+    # environment: # TODO: Have a VirtualiZarr maintainer create a test-release environment
```
Comment (TomNicholas) [inline .github/workflows/release.yml:54]: What needs to be in a `test-release` environment? Are the unit test going to be run in there?

Comment (TomNicholas) [reply .github/workflows/release.yml:54]: Okay. Is this different from a conda environment?

**Comment by TomNicholas** [comment]
@maxrjones feel free to merge/push whatever you want here / ping me for anything specific. I trust your judgement and appreciate the help!

File: .github/workflows/release.yml (hunk 1 above)
Comment (maxrjones) [reply .github/workflows/release.yml:54]: Yeah, it's a [GitHub Actions](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) environment. Thanks for sorting out the permissions, I'll work on getting it set up

**Comment by maxrjones** [comment]
Since GitHub does not allow OIDC tokens to be issued for PRs generated from a fork, it's not possible to effectively test this PR. I'm going to close it and re-open a PR from a branch within the source repository.

**Pull request closed without merging**